### ISSUE TITLE
Implement dataset_utils that works over the RAP1 dataset

### ISSUE THREAD

**Issue by AlejandroAlonsoG** [Sawaiz-Asif/IMGEAN#17]
Get/Add/Edit/Remove labels from dataset
Fetch one image
Fetch all images
Add/ Remove one image to the dataset
Get /Add/ Edit /Remove labels of one image

| **Key**           | **Data Type**               | **Description**                                                                                                                                                     |
|-------------------|-----------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `description`     | `<class 'str'>`             | General description of the dataset, e.g., `'rap2'`.                                                                                                                 |
| `reorder`         | `<class 'str'>`             | Defines the reordering strategy, e.g., `'group_order'`.                                                                                                             |
| `root`            | `<class 'str'>`             | Path to the root directory where images are stored, e.g., `'/mnt/data1/jiajian/datasets/attribute/RAP2/RAP_dataset'`.                                                |
| `image_name`      | `<class 'list'>`            | List of image file names, e.g., `['CAM16-2014-03-27-20140327102842-20140327103442-tarid89-frame1831-line1.png', ...]`.                                               |
| `label`           | `<class 'numpy.ndarray'>`   | Array of labels for each image, e.g., `[0 0 1 0 0 ...]`.                                                                                                            |
| `attr_name`       | `<class 'list'>`            | List of attribute names corresponding to the labels, e.g., `['hs-BaldHead', ...]`.                                                                                   |
| `label_idx`       | `<class 'easydict.EasyDict'>`| Dictionary categorizing label indices into groups such as:                                                                                                           |
|                   |                             | - **eval**: Attributes                                                                                                                |
|                   |                             | - **color**: Attributes related to colors of clothing or accessories.                                                                                               |
|                   |                             | - **extra**: Additional or miscellaneous attributes.                                                                                                                |
| `partition`       | `<class 'easydict.EasyDict'>`| Dictionary specifying dataset splits:                                                                                                                               |
|                   |                             | - **train**: Indices of images in the training set.                                                                                                                 |
|                   |                             | - **val**: Indices of images in the validation set.                                                                                                                 |
|                   |                             | - **test**: Indices of images in the test set.                                                                                                                      |
|                   |                             | - **trainval**: Combined indices of the training and validation sets, typically used for cross-validation or final training.                                          |
| `weight_train`    | `<class 'list'>`            | Weights associated with the training set, typically used to handle class imbalance, e.g., `[array([6.15402637e-03, 1.93646699e-01,...]`.                             |
| `weight_trainval` | `<class 'list'>`            | Weights associated with the combined train/val set, e.g., `[array([6.15402637e-03, 1.93646699e-01,...]`.                                                             |
